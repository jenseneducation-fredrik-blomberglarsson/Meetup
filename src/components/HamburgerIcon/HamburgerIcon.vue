<template>
  <div class="hamburgerIcon">
    <nav class="burgerBtn">
      <img
        class="burgerIconImg"
        src="../../assets/navicon.svg"
        alt="burgerIconImg"
        @click="changeThisBurgerToggle"
      />
      <HamburgerMenu />
    </nav>
  </div>
</template>

<script>
import { mapActions } from "vuex";
import HamburgerMenu from "@/components/HamburgerMenu/HamburgerMenu.vue";
export default {
  name: "HamburgerIcon",
  components: {
    HamburgerMenu,
  },
  methods: {
    ...mapActions(["changeThisBurgerToggle"]),
  },
};
</script>

<style>
</style>