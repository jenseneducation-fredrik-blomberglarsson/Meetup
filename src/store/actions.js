import { getMeetups } from "../data/meetupData";

const actions = {
  async fetchMeetups(context) {
    const data = await getMeetups();
    context.commit("setMeetups", data);
  }
}

export default actions;