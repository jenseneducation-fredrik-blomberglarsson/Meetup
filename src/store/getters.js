const getters = {
  getMeetups: state => {
    return state.meetups;
  },
  getJoinedMeetups: state => {
    return state.joinedMeetups;
  },
  getPreviousMeetups: state => {
    return state.previousMeetups;
  },
  getMeetupToggle: state => {
    return state.meetupToggle;
  },
  getBurgerToggle: state => {
    return state.burgerToggle;
  }
}


export default getters;