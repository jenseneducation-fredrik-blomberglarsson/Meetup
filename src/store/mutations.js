const mutations = {
  setMeetups(state, meetups) {
    state.meetups = meetups;
  }
}

export default mutations;