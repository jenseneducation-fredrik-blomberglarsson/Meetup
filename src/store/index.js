import Vue from 'vue'
import Vuex from 'vuex'
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex)

//Global state
export default new Vuex.Store({
  state: {
    meetups: [],
    joinedMeetups: [],
    previousMeetups: [],
    meetupToggle: false,
  },

  actions: actions,
  mutations: mutations,
  getters: getters
});