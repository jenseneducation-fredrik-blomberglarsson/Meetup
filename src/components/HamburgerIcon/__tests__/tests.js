import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import store from '../../../store/index.js';
import HamburgerIcon from '@/components/HamburgerIcon/HamburgerIcon.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Hamburger button is pressed', () => {
  test('Test so that everything renders properly', async () => {
    const wrapper = shallowMount(HamburgerIcon, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  })
  test('The hamburger icon (button) should show when the page is rendered', async () => {
    const wrapper = shallowMount(HamburgerIcon, {
      store,
      localVue,
    });

    const hamburgerButton = wrapper.find('.burgerIconImg');
    expect(hamburgerButton.exists()).toBe(true);
  });

  test('The hamburger icon (button) should toggle HamburgerMenu component on click', async () => {
    const state = {
      burgerToggle: false
    };
    const actions = {
      changeThisBurgerToggle: jest.fn()
    };
    const getters = {
      getBurgerToggle() {
        return state => state.burgerToggle
      }
    };
    const store = new Vuex.Store({
      state,
      actions,
      getters
    });
    const wrapper = shallowMount(HamburgerIcon, {
      store,
      localVue
    });

    const hamburgerButton = wrapper.find('.burgerIconImg');
    await hamburgerButton.trigger('click');
    expect(actions.changeThisBurgerToggle).toHaveBeenCalled();
  })
})