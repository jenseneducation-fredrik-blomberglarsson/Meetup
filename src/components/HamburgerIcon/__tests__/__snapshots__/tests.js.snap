// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger button is pressed Test so that everything renders properly 1`] = `
<div
  class="hamburgerIcon"
>
  <nav
    class="burgerBtn"
  >
    <img
      alt="burgerIconImg"
      class="burgerIconImg"
      src="../../assets/navicon.svg"
    />
     
    <hamburgermenu-stub />
  </nav>
</div>
`;
